soem/ethercatbase.c

```diff
@@ -142,7 +142,7 @@ int ecx_adddatagram(ecx_portt *port, void *frame, uint8 com, uint8 idx, boolean
 
    /* return offset to data in rx frame
       14 bytes smaller than tx frame due to stripping of ethernet header */
-   return prevlength + EC_HEADERSIZE - EC_ELENGTHSIZE - ETH_HEADERSIZE;
+   return (uint16)(prevlength + EC_HEADERSIZE - EC_ELENGTHSIZE - ETH_HEADERSIZE);
```

Comment:
(uint16) cast should not be needed anymore

Reply:
yes, if i made prevlength an int. So perhaps this is a better fix

* I reverted prevlength to be "uint16 prevlength;"
* And added an explicit typecast that might be more obvious, prevlength = (uint16)port->txbuflength[idx];

Reply:
fixed